services/harmony/app/middleware/shapefile-converter.ts

```diff
@@ -81,6 +84,103 @@ const contentTypesToConverters = {
   'application/shapefile+zip': { name: 'ESRI Shapefile', geoJsonConverter: _esriToGeoJson },
 };
 
+/**
+ *
+ * @param lon - longitude
+ * @returns
+ */
+function normalizeLongitude(lon: number): number {
+  while (lon > 180) lon -= 360;
+  while (lon < -180) lon += 360;
+  return lon;
+}
+
+/**
+ *
+ * @param geojson - normalize all the longitudes in the file to [-180,180]
+ * @returns
+ */
+export function normalizeGeoJsonCoords(geojson: any): any {
+  function normalizeCoordinates(coordinates: any): any {
+    if (Array.isArray(coordinates[0])) {
+      return coordinates.map(normalizeCoordinates);
+    } else {
+      return [normalizeLongitude(coordinates[0]), coordinates[1]];
+    }
+  }
+
+  function normalizeGeometry(geometry: any): any {
```

Comment:
I think you might be able to specify a type here from the GeoJSON libraries. Worst case we can define an interface with `type`, `coordinates`, and `geometries`.

Reply:
This is chatGPT black box wizardry that I'm not really wanting to mess around with. I definitely don't see any benefit to adding types for it, but we can discuss if you prefer to add types.

Reply:
Types are just documentation - we try to limit the use of `any` as much as possible. I don't see a reason for ignoring types here when we know what type it supports and returns. If we really want to use `any` here, I think we should get rid of the ignore types at the top of the file and put it above each line we want to ignore types for.

Reply:
Regarding types, if we were going to use them somewhere else, then sure, I would add them. But in this case, we are just  passing the geojson as an `object` and checking various fields. Creating a type that captures all of geojson  seems like overkill. This function is not exported so is not part of some external API. We could change `any` to `object` if you prefer - again, this is just what chatGPT came up with. 

Regarding the single line vs entire file linter disable, I had to use the entire file disable, because I'm already using a "disable for next line" linter disable comment for JSDocs for the internal function. ESLint gets confused if you have two consecutive "disable for next line" comments.